pkg/blobstore/spanner_blob_access.go

```diff
@@ -0,0 +1,975 @@
+		})
+	spannerReftimeUpdateFailedCount = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Namespace: "buildbarn",
+			Subsystem: "blobstore",
+			Name:      "spanner_reftime_update_failed_total",
+			Help:      "Number of spanner object reference times updates failed",
+		})
+	spannerExpiredBlobReadIgnoredCount = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Namespace: "buildbarn",
+			Subsystem: "blobstore",
+			Name:      "spanner_expired_blob_read_ignored_total",
+			Help:      "Number of ignored read attempts of expired spanner blobs",
+		})
+	gcsFailedReadDeletedBlobCount = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Namespace: "buildbarn",
+			Subsystem: "blobstore",
+			Name:      "gcs_failed_read_deleted_blob_total",
+			Help:      "Number of deletes of GCS blobs that couldn't be read",
+		})
+	gcsFailedReadDeleteBlobFailedCount = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Namespace: "buildbarn",
+			Subsystem: "blobstore",
+			Name:      "gcs_failed_read_delete_blob_failed_total",
+			Help:      "Number of failed deletes of GCS blobs that couldn't be read",
+		})
+	gcsPutFailedContextCanceledCount = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Namespace: "buildbarn",
+			Subsystem: "blobstore",
+			Name:      "gcs_put_failed_context_canceled_total",
+			Help:      "Number of failed puts of GCS blobs because the context was canceled",
+		})
+	gcsPutFailedDeadlineExceededCount = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Namespace: "buildbarn",
+			Subsystem: "blobstore",
+			Name:      "gcs_put_failed_deadline_exceeded_total",
+			Help:      "Number of failed puts of GCS blobs because the deadline was exceeded",
+		})
+	gcsPutFailedOtherCount = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Namespace: "buildbarn",
+			Subsystem: "blobstore",
+			Name:      "gcs_put_failed_other_total",
+			Help:      "Number of failed puts of GCS blobs because of other reasons",
+		})
+	spannerMalformedBlobDeletedCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "buildbarn",
+			Subsystem: "blobstore",
+			Name:      "spanner_malformed_blob_deleted_total",
+			Help:      "Number of malformed blobs that were deleted",
+		},
+		[]string{"backend_type"})
+	spannerMalformedBlobDeleteFailedCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "buildbarn",
+			Subsystem: "blobstore",
+			Name:      "spanner_malformed_blob_delete_failed_total",
+			Help:      "Number of malformed blobs that could not be deleted",
+		},
+		[]string{"backend_type"})
+	backendOperationsDurationSeconds = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: "buildbarn",
+			Subsystem: "blobstore",
+			Name:      "backend_operations_duration_seconds",
+			Help:      "Amount of time spent per backend operation in seconds.",
+			Buckets:   util.DecimalExponentialBuckets(-3, 6, 2),
+		},
+		[]string{"storage_type", "backend_type", "operation"})
```

Comment:
Isn't this metric superfluous? We already provide such a metric as part of MetricsBlobAccess.

Reply:
Does that give me enough visibility to distinguish between Spanner and GCS accesses?